components/eam/src/physics/cam/hetfrz_classnuc_cam.F90

```diff
@@ -876,368 +890,383 @@ end subroutine hetfrz_classnuc_cam_save_cbaero
-                       aer(ii,kk,mom_accum)*mom_equivso4_factor/specdens_mom + &
-                       aer(ii,kk,soa_accum)*soa_equivso4_factor/specdens_soa )/rhoair
+!====================================================================================================                                                                                                                                      
+subroutine calculate_mass_mean_radius(ii, kk, ncnst, aer, &
+                                      total_interstitial_aer_num, &
+                                      hetraer)
 
+   !***************************************************
+   ! calculate mass mean radius for BC and dust
+   !***************************************************
 
-      vol_core(2) = aer(ii,kk,dst_accum)/(specdens_dust*rhoair)
+   ! input
+   integer,  intent(in) :: ii, kk, ncnst
+   real(r8), intent(in) :: aer(pcols,pver,ncnst)         ! interstitial aerosol concentrations [kg/m^3]
+   real(r8), intent(in) :: total_interstitial_aer_num(3) ! interstitial concentrations of BC and dust [#/cm^3]
 
-      !   ratio1 = vol_shell/vol_core = 
-      !      actual hygroscopic-shell-volume/dust-core-volume
-      !   ratio2 = 6.0_r8*dr_so4_monolayers_pcage/(dgncur_a*fac_volsfc_dust)
-      !      = (shell-volume corresponding to n_so4_monolayers_pcage)/core-volume 
-      !      The 6.0/(dgncur_a*fac_volsfc_dust) = (mode-surface-area/mode-volume)
-      !   Note that vol_shell includes both so4, pom, AND soa as "equivalent so4",
-      !      The soa_equivso4_factor accounts for the lower hygroscopicity of soa.
-      !
-      !   Define xferfrac_pcage = min( 1.0, ratio1/ratio2)
-      !   But ratio1/ratio2 == tmp1/tmp2, and coding below avoids possible overflow 
+   ! output
+   real(r8), intent(out) :: hetraer(3)                   ! BC and Dust mass mean radius [m]
 
-      ! bc
+   ! local variables
+   real(r8) :: bc_num, dst1_num, dst3_num                ! number concentration [#/cm^3]
+   real(r8), parameter :: num_cm3_to_m3  = 1.0e6_r8          ! volume unit conversion, #/cm^3 to #/m^3
+   real(r8), parameter :: aermc_min_threshold  = 1.0e-30_r8
+   real(r8), parameter :: aernum_min_threshold = 1.0e-3_r8
+   real(r8), parameter :: r_bc_prescribed = 0.067e-6_r8
+   real(r8), parameter :: r_dust_a1_prescribed = 0.258e-6_r8
+   real(r8), parameter :: r_dust_a3_prescribed = 1.576e-6_r8
+
+   bc_num   = total_interstitial_aer_num(1)
+   dst1_num = total_interstitial_aer_num(2)
+   dst3_num = total_interstitial_aer_num(3)
+
+   hetraer(1) = r_bc_prescribed
+   hetraer(2) = r_dust_a1_prescribed
+   hetraer(3) = r_dust_a3_prescribed
+
+   ! BC
+   if ((aer(ii,kk,bc_accum)+aer(ii,kk,bc_pcarbon))*1.0e-3_r8 > aermc_min_threshold .and. bc_num > aernum_min_threshold) then
+      hetraer(1) = (3._r8/(4*pi*specdens_bc)*(aer(ii,kk,bc_accum)+aer(ii,kk,bc_pcarbon))/ &
+                   (bc_num*num_cm3_to_m3))**(1._r8/3._r8)
```

Comment:
Can we create a function for:
```
(3._r8/(4*pi*specdens_bc)*(aer(ii,kk,bc_accum)+aer(ii,kk,bc_pcarbon))/ &
                   (bc_num*num_cm3_to_m3))**(1._r8/3._r8)
```
as the same calculation is invoked for the following "if" conditions as well.

Also, make `hetraer`=0 so that it has a value if these "if" conditions are not triggered. `hetraer` is an intent-out variable.

Reply:
made a function get_aer_radius for the calculation.
hetraer is initialized with prescribed radius values.